Tests/Tests/NSArray_TDTAdditions_Tests.m

```diff
@@ -67,4 +68,37 @@ - (void)testReversing {
   XCTAssertEqualObjects(expected, result);
 }
 
+- (NSDictionary *)histogramForArray:(NSArray *)array {
+  __block NSMutableDictionary *histogram = [NSMutableDictionary dictionary];
+  [array tdt_applyBlock:^(id obj) {
+    if (histogram[obj] == nil) {
+      histogram[obj] = @0;
+    }
+    histogram[obj] = @([histogram[obj] longValue] + 1);
+  }];
+  return histogram;
+}
+
+- (BOOL)array:(NSArray *)array1 hasSameElementsAsInArray:(NSArray *)array2 {
+  return [[self histogramForArray:array1] isEqual:[self histogramForArray:array2]];
+}
+
+- (void)testShuffling {
+  NSArray *receiver = @[@1, @2, @3, @4, @5, @6, @7, @8];
+  NSArray *result1 = [receiver tdt_shuffledArray];
+  NSArray *result2 = [receiver tdt_shuffledArray];
+  XCTAssertTrue([self array:receiver hasSameElementsAsInArray:result1]);
+  XCTAssertTrue([self array:receiver hasSameElementsAsInArray:result2]);
+  BOOL shuffledArraysAreDifferent = ![result1 isEqual:receiver] && ![result2 isEqual:result1];
+  for (NSUInteger i = 0; i < 100; i++) {
+    if (shuffledArraysAreDifferent) {
+      break;
+    }
+    result1 = [receiver tdt_shuffledArray];
+    result2 = [receiver tdt_shuffledArray];
+    shuffledArraysAreDifferent = ![result1 isEqual:receiver] && ![result2 isEqual:result1];
+  }
+  XCTAssertTrue(shuffledArraysAreDifferent);
```

Comment:
You should split this test case into multiple tests. Idea is test case name can represent what you intended to test. For example here:
- testShuffledArrayIsDifferentThanOriginalForSameElements
- testTwoShuffledArraysAreDifferentForSameElements

While first test will always pass, second has a probability to fail. I guess we can live with that.